### PR TITLE
Migrate vscode to @types/vscode and vscode-test

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -481,8 +481,7 @@
     "compile": "tsc -p . && webpack --config webpack.config.js",
     "watch": "webpack --config webpack.config.js --watch --info-verbosity verbose",
     "build": "webpack --config webpack.config.js --mode=\"production\"",
-    "postinstall": "node ./node_modules/vscode/bin/install",
-    "test": "node ./scripts/download-vscode-for-system-tests && node ./scripts/install-vsix-dependencies redhat.java && node ./scripts/install-vsix-dependencies vscode-java-debug-0.20.0.vsix && node ./scripts/run-vscode-tests"
+    "test": "tsc -p . && node ./out/test/index.js"
```
Comment (jdneo) [inline package.json]: How to install the dependencies?

Comment (testforstephen) [reply package.json]: +1

How to run it in cli?

Comment (Eskibear) [reply package.json]: I'm following the [official guide](https://code.visualstudio.com/api/working-with-extensions/testing-extension#custom-setup-with-vscodetest) to install redhat.java before running the test cases. 
But as you can see from the CI log, the `vscodeExecutablePath` does NOT point to the code CLI as expected, thus cannot successfully install the extension.

**Comment by Eskibear** [comment]
external dependency of vscode-test issue  https://github.com/microsoft/vscode-test/issues/29 .

**Comment by Eskibear** [comment]
CI passed now, but actually the test cases are skipped.
Do NOT merge until https://github.com/microsoft/vscode-test/issues/31 get resolved.

**Pull request merged**